lib/parser.py

```diff
@@ -35,27 +35,30 @@ def searchAcrossAll(self, keyword, workflow):
     def searchInSheet(self, keyword, sheetName, workflow):
         return self.filter(self.__parseSheet(sheetName), keyword, workflow)
 
-    def __parseSheet(self, filename):
+    def __parseSheet(self, filename, specify_sheet=False):
         with open(self._sheetMapping.get(filename), 'r') as f:
-            content=f.read().decode('utf-8',"replace").strip()
-        # Tokenize to get each "item" by spliting the "\n\n". This rule must be repspected
-        content=[item.strip() for item in content.split("\n\n")]
-        # ASSUME the item pattern is "comment, comment, comment ..., command"
-        # An item doesn't have to contain comments but must have a command
+            content=f.read().decode('utf-8',"replace").strip("\n")
+        # Tokenize to get each "item" by spliting the "\n\n" (possibly with white space characters).
+        # This rule must be respected
+        content=re.split("\n\s*\n",content)
+        content=[item.strip() for item in content]
         items=[]
         for item in content:
-            try:
-                comment, command=item.rsplit("\n",1)
-            except ValueError:
-                continue
-            # Or you wanna see which line doesn't comform to the format
-            #   comment=""
-            #   command="Fail to parse: {}".format(item)
-            # cleanup "#" and \n
-            comment=comment.replace("#","").replace("\n",". ").strip()
-            command=command.strip()
-            items.append((comment, command))
-        return [dict(comment=comment, command=command) for comment,command in items] # [{}, {}, {}...]
+            if item[0:2]=="/*":
+                command=""
+            else:
+                lines=item.split("\n")
+                comment_list=[line[1:] for line in lines if line.startswith('#')]
+                command_list=[line for line in lines if not line.startswith('#')]
+                comment=", ".join(comment_list)
+                command=u"  \u28B8  ".join(command_list)
```

Comment:
Such rules may allow 
```
# comment one
# comment two
accidental line without # 
# comment three
My Command
```

to be parsed as 

```
commandlist = ['My Command', 'accidental line ....'].
```

The order might be messed up. Sure this shouldn't be a problem as a user wouldn't write like this I guess. Just wanna confirm you're aware of this thing.

Reply:
And it allows multiple comments and multiple commands as "one node" right? is this your design?

Reply:
Yes, it allows multiple comments and multiple commands as one node. I agree that a standard use would be to either have multiple comments and one command or one comment with multiple commands, but I am not sure that it is worth ruling out other possibilities. What do you think?

Concerning your example with an accidental line, you are right. Do you think that it would be better in that case to stop after the accidental line and simply ignore what comes next? It is not difficult to implement and I have no clear opinion on the subject.

Reply:
1. Yeah the design is good actually. Allowing multiple comments + multiple command separated by "|" = one node solves some problems (e.g., similar commands can tag with same comments now). It's definitely helpful. Just stick to it :)

2. How about we just ignore the "accidental line"? like it never exists. Hence if the accidental line is the last line of the node we treat it as command anyway.  but if we encounter the pattern of "#, #, line, #, command", we simply parse it as "#, #, #, command". What do you think?

I feel this "parsing rule" a little bit stupid but ... Alfred is just what it is lol.